### PR TITLE
Prepare for version 3

### PR DESCRIPTION
**Pull request by weierophinney**
This patch prepares the package for version 3. Specifically:
- Updates the minimum required PHP version to 5.6.
- Updates to the latest stable release of existing packages.
- Adds new v3-specific components:
  - zend-mvc-console
  - zend-mvc-plugins
  - zend-mvc-form
  - zend-mvc-i18n
  - zend-json-server
  - zend-xml2json
  - zend-diactoros
  - zend-stratigility
  - zend-psr7bridge
  - zend-hydrator
  - zend-servicemanager-di
- Updates the zend-ldap suggestion to specify a known-good stable version of the component.
- Removes the various `bin/*` scripts. The plugin map and class map generators are no longer relevant, and the template map generator is now part of zend-view.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,80 +5,89 @@
...
-- [zend-tag](https://github.com/zendframework/zend-tag)
-- [zend-test](https://github.com/zendframework/zend-test)
-- [zend-text](https://github.com/zendframework/zend-text)
-- [zend-uri](https://github.com/zendframework/zend-uri)
-- [zend-validator](https://github.com/zendframework/zend-validator)
-- [zend-version](https://github.com/zendframework/zend-version)
-- [zend-view](https://github.com/zendframework/zend-view)
-- [zend-xmlrpc](https://github.com/zendframework/zend-xmlrpc)
-- [ZendXml](https://github.com/zendframework/ZendXml)
+- [zendframework/zend-authentication](https://github.com/zendframework/zendframework/zend-authentication)
```
Comment (akrabat) [inline CONTRIBUTING.md]: Too many `zendframework`s here!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,80 +5,89 @@
...
+- [zendframework/zendxml](https://github.com/zendframework/zendframework/zendxml)
 
 ## Reporting Potential Security Issues
 
-If you have encountered a potential security vulnerability, please **DO NOT** report it on the public
-issue tracker: send it to us at [zf-security@zend.com](mailto:zf-security@zend.com) instead.
-We will work with you to verify the vulnerability and patch it as soon as possible.
+If you have encountered a potential security vulnerability, please **DO NOT**
+report it on public issue trackers; instead, send it to us at
+[zf-security@zend.com](mailto:zf-security@zend.com) instead.  We will work with
```
Comment (akrabat) [inline CONTRIBUTING.md]: Don't need double spaces after a full stop in this day an age :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (weierophinney) [inline CONTRIBUTING.md]: Fixed; thanks, @akrabat

File: CONTRIBUTING.md (hunk 2 above)
Comment (weierophinney) [inline CONTRIBUTING.md]: Surprisingly, that was `vim`'s auto-formatting!

File: README.md [hunk 3]
```diff
@@ -1,26 +1,37 @@
...
 
 ### SYSTEM REQUIREMENTS
 
-Zend Framework 2 requires PHP 5.5 or later; we recommend using the
+Zend Framework 3 requires PHP 5.6 or later; we recommend using the
 latest PHP version whenever possible.
 
 ### INSTALLATION
 
+We no longer recommend installing this package directly. The package is a
```
Comment (akrabat) [inline README.md]: I'd like to see the recommended installation process noted here - presumably we recommend starting with the skeleton?

File: README.md [hunk 4]
```diff
@@ -30,7 +41,7 @@ $ composer require zendframework/zendframework
 A great place to get up-to-speed quickly is the Zend Framework
 QuickStart:
 
-http://framework.zend.com/manual/current/en/user-guide/overview.html
+- https://docs.zendframework.com/tutorials/getting-started/overview/
```
Comment (akrabat) [inline README.md]: Why not link the "Zend Framework QuickStart" text above?

**Comment by akrabat** [comment]
Reviewed. Good to go now. 👍

File: composer.json [hunk 5]
```diff
@@ -1,79 +1,85 @@
 {
     "name": "zendframework/zendframework",
-    "description": "Zend Framework 2",
-    "type": "library",
+    "description": "Zend Framework",
+    "type": "metapackage",
```
Comment (akrabat) [inline composer.json]: Do we need to update .gitattributes to ensure that we don't install anything or is composer not that strict?

File: README.md (hunk 3 above)
Comment (weierophinney) [inline README.md]: Added!

File: composer.json (hunk 5 above)
Comment (weierophinney) [inline composer.json]: Nope - just confirmed that when you install a metapackage, nothing related to the package is installed in the `vendor/` directory — only the packages listed as requirements are installed. As such, the end-user will have no files from this repository installed at all. :smile:

**Comment by mwillbanks** [comment]
LGTM... although, how about a migration guide?

File: README.md [hunk 6]
```diff
@@ -72,63 +96,69 @@ This package is a metapackage aggregating the following components:
...
 - [zend-test](https://github.com/zendframework/zend-test)
 - [zend-text](https://github.com/zendframework/zend-text)
 - [zend-uri](https://github.com/zendframework/zend-uri)
 - [zend-validator](https://github.com/zendframework/zend-validator)
-- [zend-version](https://github.com/zendframework/zend-version)
 - [zend-view](https://github.com/zendframework/zend-view)
+- [zend-xml2json](https://github.com/zendframework/zend-xml2json)
 - [zend-xmlrpc](https://github.com/zendframework/zend-xmlrpc)
-- [ZendXml](https://github.com/zendframework/ZendXml)
+- [zendxml](https://github.com/zendframework/zendxml)
```
Comment (mwillbanks) [inline README.md]: Add a section on migration...

```markdown
### Migrating from v2 to v3

Zend Framework v2 to v3 has been intended as an incremental upgrade. We have even made efforts in the past year to provide forwards compatibility features in v2 versions of components, to allow users to prepare their code for upgrade.

[View the migration guide](https://docs.zendframework.com/tutorials/migration/to-v3/overview/).
```

**Pull request merged**